includes/class-woocommerce.php

```diff
@@ -109,6 +109,24 @@ final class WooCommerce {
 	 */
 	public $deprecated_hook_handlers = array();
 
+	/**
+	 * The shared instance of the dependency injection container.
+	 *
+	 * @var \Automattic\WooCommerce\Container
+	 */
+	private static $container;
+
+	/**
+	 * The list of service providers used to initialize the dependency injection container,
+	 * as names of classes in the Automattic\WooCommerce\Tools\DependencyManagement\ServiceProviders namespaced
+	 * and without the 'ServiceProvider' suffix.
+	 *
+	 * @var string[]
+	 */
+	private static $service_providers = array(
+		'Queue',
```

Comment:
What about just having `\Automattic\WooCommerce\Tools\DependencyManagement\ServiceProviders\QueueServiceProvider::class` and then in the `addServiceProvider` call below you can simply use `$provider`? This also has the nice effect of support providers in different directories!

Reply:
Yep, that's a good idea.